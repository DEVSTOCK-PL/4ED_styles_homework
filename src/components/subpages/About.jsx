import styled from "styled-components";

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  p {
    text-align: center;
  }
`;

const Title = styled.h1`
  font-size: 36px;
  text-align: center;
`;

const About = () => (
  <Wrapper>
    <Title>About</Title>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quaerat quia
      fugiat repellendus maxime soluta iusto quod distinctio possimus asperiores
      voluptas. Veniam nulla ad perferendis explicabo magnam unde aperiam labore
      in incidunt exercitationem, ex quo libero amet, distinctio quia, odit
      suscipit harum delectus. Quasi ipsam hic doloremque officiis eos non
      animi!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quaerat quia
      fugiat repellendus maxime soluta iusto quod distinctio possimus asperiores
      voluptas. Veniam nulla ad perferendis explicabo magnam unde aperiam labore
      in incidunt exercitationem, ex quo libero amet, distinctio quia, odit
      suscipit harum delectus. Quasi ipsam hic doloremque officiis eos non
      animi! Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptates
      adipisci blanditiis praesentium debitis cumque libero fuga ullam animi
      sint eligendi.
    </p>
  </Wrapper>
);
export default About;
