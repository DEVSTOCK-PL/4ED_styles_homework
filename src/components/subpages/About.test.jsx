import { test, describe, expect } from "vitest";

import { fireEvent, render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";

import About from "./About";

describe("About Component", () => {
  //testujemy czy komponent się wyświetla z stanem początkowym
  test("renders without crash", () => {
    //renderujemy about
    render(<About />);
    //sprawdzamy stan początkowy - licznik:0
    expect(screen.getByText(/Licznik: 0/)).toBeInTheDocument();
  });

  test("increments counter on btn click", () => {
    render(<About />);
    fireEvent.click(screen.getByText("+"));
    expect(screen.getByText(/Licznik: 1/)).toBeInTheDocument();
  });

  test("decrements counter on btn click", () => {
    render(<About />);
    fireEvent.click(screen.getByText("-"));
    expect(screen.getByText(/Licznik: -1/)).toBeInTheDocument();
  });

  test("resets counter on btn click", () => {
    render(<About />);
    fireEvent.click(screen.getByText("+"));
    fireEvent.click(screen.getByText("reset"));
    expect(screen.getByText(/Licznik: 0/)).toBeInTheDocument();
  });
  test("multiple changes on counter on btn click", () => {
    render(<About />);
    fireEvent.click(screen.getByText("+"));
    fireEvent.click(screen.getByText("+"));
    fireEvent.click(screen.getByText("+"));
    fireEvent.click(screen.getByText("-"));
    expect(screen.getByText(/Licznik: 2/)).toBeInTheDocument();
  });
  test("update state on input change", () => {
    render(<About />);
    const input = screen.getByRole("textbox");
    fireEvent.change(input, { target: { value: "new value" } });
    expect(input.value).toBe("new value");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });
});
