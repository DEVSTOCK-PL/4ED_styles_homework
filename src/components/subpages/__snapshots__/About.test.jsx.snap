// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beyTiQ iFsJpF"
  >
    <div>
      Licznik: 0
      <button
        class="w-8 h-8 m-4 bg-slate-400"
      >
        +
      </button>
      <button
        class="w-8 h-8 m-4 bg-slate-400"
      >
        -
      </button>
      <button
        class="w-16 h-8 m-4 bg-slate-400"
      >
        reset
      </button>
    </div>
    <input
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
